cobc/field.c

```diff
@@ -2893,10 +2893,13 @@ validate_field_value (struct cb_field *f)
 	if (f->values) {
 		if (f->flag_picture_l) {
 			cb_error_x (CB_TREE (f),
-				    _("field with PICTURE L may not have "
-				      "a VALUE clause"));
+				    _("%s and %s are mutually exclusive"),
+				    "PICTURE L", "VALUE");
+			cobc_parse_free (f->values);
```

Comment:
The idea is good and reasonable. Some people will say "that's not relevant" because at the end of the parsing the values will be freed in any case (that's the purpose of the parse malloc); but _if_ you opt do do it then that needs to be done completely (the f->values is always a list with at least (in this case only) one contained cb_tree - only freeing the list is not good.
I think I'd personally leave that to the auto-cleanup at the end of parsing...

Reply:
Ok (I though the free was greedier, but I should have checked).